src/main/java/com/intuit/async/execution/util/PredicateEvaluator.java

```diff
@@ -0,0 +1,36 @@
+package com.intuit.async.execution.util;
+
+import java.util.function.Predicate;
+
+
+/**
+ * @param <T>
+ * @author jatin14493
+ * Generic PredicateEvaluator class
+ */
+public class PredicateEvaluator<T> {
```

Comment:
why do we need to have a PredicateEvaluator class ? 
new RxExecutionChain(req, task).next(task1,testCondn()).execute();

We can something like this
Predicate<Map<String, Object>> cond = testCondn();
cond.test(new HashMap<>());

private Predicate<Map<String, Object>> testCondn() {
		return x -> x.containsKey("abx");
}